src/cols.py

```diff
@@ -0,0 +1,77 @@
+import re
+from  src.Num import Num
+from  src.sym import Sym
+
+'''
+Class `Cols` Summarizes the first row of the dataset
+'''
+
+class Cols:
+    '''
+    Constuctor which initializes the attributes
+
+    Attributes:
+    @names: column names
+    @all: storing Num or Sym objects
+    @klass: the single dependent class if it exists
+    @x: independent columns
+    @y: dependent columns
+
+    Arguments:
+    @names: column names
+    '''
+    def __init__(self, names) -> None:
+        self.names = names 
+        self.all = []
+        self.klass = None
+        self.x = []
+        self.y = []
+
+        '''
+        First, storing all the Num and Sym Objects into the "all" attribute
+        Second, performing regex to separate the dependent from independent variables into the attributes "y" and "x"
+        respectively
+        For the atributes "x" and "y" we skip all columns which end with ":".
+        '''
+        for i in range(len(names)):
+            if(re.search("\*[A-Z]", names[i])):
+                self.all.append(Num(i, names[i]))
```

Comment:
Can you check in the code file of Lua, over there they are using the push function which they have written manually. So are we supposed to use the append function? Or do we have to implement a push function?

Reply:
Im guessing we can use this?
In LUA we have only one kind of list(dictionary basically). The push function is just using array indexing so i think it should be fine.

Reply:
Okay!!